File: fastapi/src/schemas/common_schemas.py
from pydantic import BaseModel, Field


class ErrorResponse(BaseModel):
    error: str = Field(description="Error message", examples=["Something went wrong"])


class HealthResponse(BaseModel):
    status: str = Field(examples=["ok"])
    timestamp: str = Field(description="Current server timestamp")
    services: dict[str, str] = Field(examples=[{"database": "connected", "storage": "connected"}])

File: hono-bun/src/schemas/common-schemas.ts
import { z } from "@hono/zod-openapi";

export const ErrorSchema = z.object({
  error: z.string().openapi({
    description: "Error message",
    example: "Something went wrong",
  }),
});

export const HealthResponseSchema = z.object({
  status: z.string().openapi({ example: "ok" }),
  timestamp: z.string().datetime().openapi({
    description: "Current server timestamp",
  }),
  services: z.object({
    database: z.string().openapi({ example: "connected" }),
    storage: z.string().openapi({ example: "connected" }),
  }),
});
